src/components/PaginationFooter/Dropdowns/DropDown.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import type { SelectChangeEvent } from '@mui/material/Select';
+import { usePaginationContext } from '../context/PaginationProvider';
+import { StyledSelect, StyledTypography } from '../styles';
+
+interface Props {
+  name: string;
+  value: number;
+  options: JSX.Element;
+  handleChange: (event: SelectChangeEvent<number>) => void;
+}
+
+const DropDown = ({ name, value, options, handleChange }: Props) => {
+  const { layout, footerStyle, translator, tabIndex } = usePaginationContext();
+  const chartId = layout.qInfo.qId as string;
+  const label = translator.get(`NebulaTableUtils.Pagination.${name}`);
+  const baseId = `${name}-dropdown`;
+  const labelId = `${baseId}-label-${chartId}`;
+  const id = `${baseId}-${chartId}`;
+  const inputProps = {
+    tabIndex,
+    id,
+    'data-testid': baseId,
+    'aria-labelledby': labelId,
+  };
```

Comment:
I think a useMemo wrapper for all of this variables would be optimal

Reply:
why? They are super cheap to recalculate? I mean the component will rerender anyway right?

Reply:
I will do some debugging to see when the whole pagination footer re-renders. I should only be if some of the props have changed and that should not happen all that often. My guess is that it only happens when the whole chart have to re-render anyway.

Reply:
I agree with you that they are cheap to recalculate but,
`<DropDown />` Component uses Layout which comes as a prop to pagination -> so i assume everything changes in layout it will trigger a rerender

Reply:
Yes, but if the `layout` changes the whole chart have to re-render anyway